conan/tools/env/virtualrunenv.py

```diff
@@ -10,11 +10,14 @@ def runenv_from_cpp_info(conanfile, cpp_info):
     if cpp_info.bin_paths:  # cpp_info.exes is not defined yet
         dyn_runenv.prepend_path("PATH", cpp_info.bin_paths)
     # If it is a build_require this will be the build-os, otherwise it will be the host-os
-    if cpp_info.lib_paths:
-        dyn_runenv.prepend_path("LD_LIBRARY_PATH", cpp_info.lib_paths)
-        dyn_runenv.prepend_path("DYLD_LIBRARY_PATH", cpp_info.lib_paths)
-    if cpp_info.framework_paths:
-        dyn_runenv.prepend_path("DYLD_FRAMEWORK_PATH", cpp_info.framework_paths)
+
+    os = str(conanfile.settings.get_safe("os"))
```

Comment:
Actually, it is a bit more subtle than this:

- Build context should get the ``settings_build.os``
- Host context should get the ``settings.os``